strings/base_coroutine_threadpool.h

```diff
@@ -21,16 +21,14 @@ namespace winrt::impl
         return (0 == WINRT_IMPL_CoGetApartmentType(&aptType, &aptTypeQualifier)) && ((aptType == 0 /*APTTYPE_STA*/) || (aptType == 3 /*APTTYPE_MAINSTA*/));
     }
 
-    inline bool requires_apartment_context() noexcept
+    inline auto current_context()
     {
-        int32_t aptType;
-        int32_t aptTypeQualifier;
-        return (0 == WINRT_IMPL_CoGetApartmentType(&aptType, &aptTypeQualifier)) && ((aptType == 0 /*APTTYPE_STA*/) || (aptType == 2 /*APTTYPE_NA*/) || (aptType == 3 /*APTTYPE_MAINSTA*/));
+        return capture<IContextCallback>(WINRT_IMPL_CoGetObjectContext);
```

Comment:
Note that this change means that you can no longer use the `winrt::apartment_context()` function or `co_await` without first initializing COM. The previous code forgave you and assumed that if COM was not initialized, then you were in the default MTA.

Reply:
seems ok to me